reducers/index.js

```diff
@@ -5,26 +5,28 @@ module.exports = function navigate(state, {type, payload}) {
     return {};
   }
 
-  const stack = state.stack;
-  const index = state.index;
+  const stack = state.__navRedux.stack;
+  const index = state.__navRedux.index;
 
   switch (type) {
     case PUSH:
       return Object.assign({}, state, {
-        stack: stack.unshift(payload),
-        index: 0,
+        __navRedux: {
+          stack: stack.push(payload),
+          index: 0,
+        },
       });
 
     case POP:
-      const incremented = index + 1;
-
-      if (incremented === stack.count()) {
+      if (stack.count() === 1) {
         return state;
```

Comment:
I think it's fine to throw if somebody trying to pop the initial route. 
Also, the expected behavior when you `pop` your navigation stack is to go one step back.

I don't think we need `canBack` or `canForward`, it's not that hard to guess how to implement it in the app's logic.